src/DiagnosticProvider.ts

```diff
@@ -16,22 +20,44 @@ class DiagnosticProvider {
         const dependencyCollection = vscode.languages.createDiagnosticCollection("Dependency");
         this._collection = dependencyCollection;
         context.subscriptions.push(this._collection);
-        context.subscriptions.push(vscode.workspace.onDidCloseTextDocument(doc => {
-            dependencyCollection.delete(doc.uri);
-            this.map.clear();
+        context.subscriptions.push(vscode.workspace.onDidChangeTextDocument(
+        async e => {
+            if (e.document.fileName.endsWith("pom.xml")) {
+                await this.debouncedRefresh(e.document.uri);
+            }
         }));
     }
 
-    public async refreshDiagnostics(uri: vscode.Uri, conflictNodes: Dependency[]): Promise<vscode.Diagnostic[]> {
-        this.map.clear();
```

Comment:
should not clear map because refresh diagnostics for one pom file should not change diagnostics of other pom files.